### ISSUE TITLE
failing to recognize some modules

### ISSUE THREAD

**Issue by dc25** [haskell/haskell-ide-engine#1069]
I'm using hie from inside vim and for the most part it seems to be working.  However at least one module (Data.Aeson) is not being recognized.  Autocomplete does not show this module as an option when I type it in and the editor shows an error saying that the module does not exist: "LSP: Could not find module Data.Aeson.  Perhaps you meant Data.Version..."

I know the module is there because I can compile the file just fine.

**Comment by lorenzo** [comment]
did you add the aeson package to your cabal or package.yaml file?

**Comment by dc25** [comment]
No.  I'm using nix and don't have either a cabal or package.yaml file.  Is hie behavior affected by the contents of these files?

**Comment by lorenzo** [comment]
Yes, it is. We use ghc-mod under the hood (as far as my understanding goes) it uses those files to understand where things are installed and what packages are available. I've never used nix, though, but my experience with using hie and a global GHC installation is that I need a cabal file at least to help  HIE understand what packages are available.
